tests/core/pyspec/eth2spec/test/eip7251/epoch_processing/test_process_pending_balance_deposits.py

```diff
@@ -97,3 +97,133 @@ def test_multiple_pending_deposits_above_churn(spec, state):
     assert state.deposit_balance_to_consume == spec.get_activation_exit_churn_limit(state) - 2 * amount
     # third deposit is still in the queue
     assert state.pending_balance_deposits == [spec.PendingBalanceDeposit(index=2, amount=amount)]
+
+
+@with_eip7251_and_later
+@spec_state_test
+def test_skipped_deposit_exiting_validator(spec, state):
+    index = 0
+    amount = spec.MIN_ACTIVATION_BALANCE
+    state.pending_balance_deposits.append(spec.PendingBalanceDeposit(index=index, amount=amount))
+    pre_pending_balance_deposits = state.pending_balance_deposits
+    pre_balance = state.balances[index]
+    # Initiate the validator's exit
+    spec.initiate_validator_exit(state, index)
+    yield from run_epoch_processing_with(spec, state, 'process_pending_balance_deposits')
+    # Deposit is skipped because validator is exiting
+    assert state.balances[index] == pre_balance
+    # All deposits either processed or postponed, no leftover deposit balance to consume
+    assert state.deposit_balance_to_consume == 0
+    # The deposit is still in the queue
+    assert state.pending_balance_deposits == pre_pending_balance_deposits
+
+
+@with_eip7251_and_later
+@spec_state_test
+def test_multiple_skipped_deposits_exiting_validators(spec, state):
+    amount = spec.EFFECTIVE_BALANCE_INCREMENT
+    for i in [0, 1, 2]:
+        # Append pending deposit for validator i
+        state.pending_balance_deposits.append(spec.PendingBalanceDeposit(index=i, amount=amount))
+
+        # Initiate the exit of validator i
+        spec.initiate_validator_exit(state, i)
+    pre_pending_balance_deposits = state.pending_balance_deposits
+    pre_balances = state.balances
+    yield from run_epoch_processing_with(spec, state, 'process_pending_balance_deposits')
+    # All deposits are postponed, no balance changes
+    assert state.balances == pre_balances
```

Comment:
Python mutable trap: `pre_balances = state.balances` makes `pre_balances` point to `state.balances` itself, therefore `assert state.balances == pre_balances` is always true. 

```suggestion
    pre_balances = state.balances.copy()
    yield from run_epoch_processing_with(spec, state, 'process_pending_balance_deposits')
    # All deposits are postponed, no balance changes
    assert state.balances == pre_balances
```

Reply:
thanks!